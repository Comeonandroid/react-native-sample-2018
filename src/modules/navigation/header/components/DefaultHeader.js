import React, { PureComponent } from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'

import CustomIcon from '../../../../utils/icon'

import { Text } from '../../../../elements'

class DefaultHeader extends PureComponent {

  constructor(params) {
    super(params)

    this.handleClick = this.handleClick.bind(this)
  }

  handleClick() {
    this.props.navigation.navigate('DrawerOpen')
  }

  render() {
    return (
      <Header className={'DefaultHeader'}>
        <Title>{this.props.title}</Title>
        <Hamburger onPress={this.handleClick}>
          <HamburgerIcon size={18} name={'menu-open'} />
        </Hamburger>
      </Header>
    )
  }
}
DefaultHeader.propTypes = {
  title: PropTypes.string.isRequired,
  navigation: PropTypes.object
}

const Header = styled.View `
  height: 80px;
  display: flex;
  background-color: ${ ({ theme }) => theme.colors.background };
  flex-direction: row;
  align-items: center;
  justify-content: center;
  padding: 0 20px;
`

const HamburgerIcon = styled(CustomIcon) `
  color: ${ ({ theme }) => theme.colors.headerContrastColor }
`

const Hamburger = styled.TouchableOpacity`
  margin-left: auto;
`

const Title = styled(Text)`
  font-size: 18px;
  align-self: center;
  margin-left: auto;
`

export default DefaultHeader
