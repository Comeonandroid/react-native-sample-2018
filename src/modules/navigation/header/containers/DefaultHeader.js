import { withTheme } from 'styled-components'
import { withNavigation } from 'react-navigation'
import { connect } from 'react-redux'
import { getCurrentRoute } from '../../../../redux/navigation'

import DefaultHeader from '../components/DefaultHeader'

const mapStateToProps = (state) => ({
  navigationState: getCurrentRoute(state)
})

export default connect(mapStateToProps)(
  withTheme(
    withNavigation(
      DefaultHeader
    )
  )
)
