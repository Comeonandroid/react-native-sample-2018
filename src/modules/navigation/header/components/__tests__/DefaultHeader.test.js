import 'react-native'
import React from 'react'

import { itBehavesLike } from '../../../../../specs/shared_examples/index'

import DefaultHeader from '../DefaultHeader'

const setup = propOverrides => {
  const props = Object.assign({
    title: 'Test',
  }, propOverrides)

  const component = <DefaultHeader {...props} />

  return {
    props,
    component
  }
}

describe('<DefaultHeader />', () => {
  const { component } = setup()
  itBehavesLike('aThemedComponent', { subject: component })
})
