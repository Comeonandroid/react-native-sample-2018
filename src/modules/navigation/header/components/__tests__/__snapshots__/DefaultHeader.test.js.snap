// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DefaultHeader /> dark: renders correctly 1`] = `
<View
  className="DefaultHeader"
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#000000",
        "display": "flex",
        "flexDirection": "row",
        "height": 80,
        "justifyContent": "center",
        "paddingBottom": 0,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 0,
      },
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#fcfcfc",
          "fontFamily": "Open Sans",
        },
        Array [
          Object {
            "alignSelf": "center",
            "fontSize": 18,
            "marginLeft": "auto",
          },
          undefined,
        ],
      ]
    }
  >
    Test
  </Text>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "marginLeft": "auto",
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 18,
          },
          Array [
            Object {
              "color": "#fcfcfc",
            },
            Array [
              Object {
                "color": "#fff",
              },
              undefined,
            ],
          ],
          Object {
            "fontFamily": "icomoon",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;

exports[`<DefaultHeader /> light: renders correctly 1`] = `
<View
  className="DefaultHeader"
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#fefaec",
        "display": "flex",
        "flexDirection": "row",
        "height": 80,
        "justifyContent": "center",
        "paddingBottom": 0,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 0,
      },
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "Open Sans",
        },
        Array [
          Object {
            "alignSelf": "center",
            "fontSize": 18,
            "marginLeft": "auto",
          },
          undefined,
        ],
      ]
    }
  >
    Test
  </Text>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "marginLeft": "auto",
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 18,
          },
          Array [
            Object {
              "color": "#4b4c4d",
            },
            Array [
              Object {
                "color": "#000",
              },
              undefined,
            ],
          ],
          Object {
            "fontFamily": "icomoon",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;
